ampd/src/asyncutil/task.rs

```diff
@@ -135,53 +177,85 @@ mod test {
-            .add_task(CancellableTask::create(waiting_task));
+            .add_task("waiting_task_1", CancellableTask::create(waiting_task))
+            .add_task("waiting_task_2", CancellableTask::create(waiting_task))
+            .add_task(
+                "immediate_task",
+                CancellableTask::create(|_| async { Ok(()) }),
+            )
+            .add_task("waiting_task_3", CancellableTask::create(waiting_task));
         assert!(tasks.run(CancellationToken::new()).await.is_ok());
     }
 
     #[tokio::test]
     async fn collect_all_errors_on_completion() {
         let tasks = TaskGroup::new("test")
-            .add_task(CancellableTask::create(|token| async move {
-                token.cancelled().await;
-                Err(report!(TaskError {}))
-            }))
-            .add_task(CancellableTask::create(|token| async move {
-                token.cancelled().await;
-                Err(report!(TaskError {}))
-            }))
-            .add_task(CancellableTask::create(|_| async { Ok(()) }))
-            .add_task(CancellableTask::create(|token| async move {
-                token.cancelled().await;
-                Err(report!(TaskError {}))
-            }))
-            .add_task(CancellableTask::create(|_| async { Ok(()) }))
-            .add_task(CancellableTask::create(|token| async move {
-                token.cancelled().await;
-                Err(report!(TaskError {}))
-            }));
+            .add_task(
+                "error_task_1",
+                CancellableTask::create(|token| async move {
+                    token.cancelled().await;
+                    Err(report!(TaskError {}))
+                }),
+            )
+            .add_task(
+                "error_task_2",
+                CancellableTask::create(|token| async move {
+                    token.cancelled().await;
+                    Err(report!(TaskError {}))
+                }),
+            )
+            .add_task(
+                "success_task_1",
+                CancellableTask::create(|_| async { Ok(()) }),
+            )
+            .add_task(
+                "error_task_3",
+                CancellableTask::create(|token| async move {
+                    token.cancelled().await;
+                    Err(report!(TaskError {}))
+                }),
+            )
+            .add_task(
+                "success_task_2",
+                CancellableTask::create(|_| async { Ok(()) }),
+            )
+            .add_task(
+                "error_task_4",
+                CancellableTask::create(|token| async move {
+                    token.cancelled().await;
+                    Err(report!(TaskError {}))
+                }),
+            );
         let result = tasks.run(CancellationToken::new()).await;
         let err = result.unwrap_err();
-        assert_eq!(err.current_frames().len(), 4);
+        assert_eq!(err.current_frames().len(), 1);
```

Comment:
while it succeeds, this test isn't very useful anymore. The point of assertions is to make sure our requirements are met. We don't care that there is only one top frame. What we do care about is that the entire error history is captured. Since all the errors are the same we don't have to worry about reordering and can use goldie to check the output